File: ゼミ用/05製造/api/src/graph-dev/request/EditTimeRequest.php
<?php

/**
 * 時系列編集処理の値を保持する
 */
class EditTimeRequest {
    private $timeId;
    private $timeName;
    private $userId;
    private $version = 0;
    private $errorMsg = array();

    /**
     * 時系列編集処理の値を格納します
     */
    public function __construct() {
    }

    public function getTimeId() {
        return $this->timeId;
    }

    public function setTimeId($timeId) {
        $this->timeId = $timeId;
    }

    public function getTimeName() {
        return $this->timeName;
    }

    public function setTimeName($timeName) {
        $this->timeName = $timeName;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function setUserId($userId) {
        $this->userId = $userId;
    }

    public function getVersion() {
        return $this->version;
    }

    public function setVersion($version) {
        $this->version = $version;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        try {
            // 入力項目チェック
            if (empty($this->timeName) || empty($this->timeId)) {
                array_push($this->errorMsg, 'エラー ：必須項目が入力されていません');
                $validationFlg = true;
            }

            // 時系列IDの文字数チェック
            if (strlen($this->timeId) != Constant::TIME_ID_DIGIT + strlen(Constant::TIME_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：時系列IDは' . Constant::TIME_ID_DIGIT + strlen(Constant::TIME_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }

            // 時系列名の文字数チェック
            if (100 < strlen($this->timeName)) {
                array_push($this->errorMsg, 'エラー ：時系列名は100文字以内で入力してください');
                $validationFlg = true;
            }

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは' . Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/EditTimeRequest.py
# /**
# * 時系列編集処理の値を保持する
# */
class EditTimeRequest:
    errorMsg = []

    # /**
    # * 時系列編集処理の値を格納します
    # */
    def __init__(self, timeId, timeName, userId, version):
        self.timeId = timeId
        self.timeName = timeName
        self.userId = userId
        self.version = version

    def getTimeId(self):
        return self.timeId

    def getTimeName(self):
        return self.timeName

    def getUserId(self):
        return self.userId

    def getVersion(self):
        return self.version

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.timeName) == 0 or len(self.timeId) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # 時系列IDの文字数チェック
            if (2000 < len(self.timeId)):
                self.errorMsg.append('エラー ：登場人物説明は1200文字以内で入力してください')
                validationFlg = True

            # 時系列名の文字数チェック
            if (100 < len(self.timeName)):
                self.errorMsg.append('エラー ：時系列名は100文字以内で入力してください')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (2000 < len(self.userId)):
                self.errorMsg.append('エラー ：登場人物説明は1200文字以内で入力してください')
                validationFlg = True      

        except Exception as e:
            validationFlg = True

        return validationFlg
